FSQLocationBroker/FSQLocationBroker.m

```diff
@@ -516,13 +516,13 @@ - (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray
     BOOL isBackgrounded = applicationIsBackgrounded();
     CLLocation *mostRecentLocation = [locations firstObject];
     for (CLLocation *location in locations) {
-        if ([mostRecentLocation.timestamp laterDate:location.timestamp] == location.timestamp) {
+        if ([mostRecentLocation.timestamp compare:location.timestamp] == NSOrderedAscending) {
```

Comment:
So this is fine, but you should also add the same nil check from below of `!mostRecentLocation ||` (and maybe even still reverse it so that you don't pass if the new `location` is `nil`, and maybe add tests that `nil` values don't mess everything up...)